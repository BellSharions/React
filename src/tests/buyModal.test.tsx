import { unmountComponentAtNode } from "react-dom";
import renderer from "react-test-renderer";
import BuyModalBody from "../components/modal/buyModalBody";

let container: Element;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
});

describe("Buy modal", () => {
  test("Snapshot test should be the same with previous version", () => {
    const mockConfirmHandler = jest.fn();
    const mockCloseHandler = jest.fn();
    const component = renderer.create(
      <BuyModalBody
        userName="test"
        cartGames={[]}
        amount={0}
        closeHandler={mockCloseHandler}
        confirmHandler={mockConfirmHandler}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
