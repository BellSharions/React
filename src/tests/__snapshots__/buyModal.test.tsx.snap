// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buy modal Snapshot test should be the same with previous version 1`] = `
<div
  className="buy__modal_container"
  role="note"
>
  <button
    className="buy__modal_close-btn"
    onClick={[MockFunction]}
    type="button"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-times fa-w-11 "
      data-icon="times"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 352 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </button>
  <div
    className="buy__modal_content-container"
  >
    <div
      className="buy__modal_contentParagraphs"
    >
      <p
        className="buy__modal_contentParagraph-user"
      >
        Dear 
        test
        , are you sure you want to buy following games for amount 
        0
        $
      </p>
    </div>
    <div
      className="buyModal_confirm-btn"
    >
      <div
        className="button__container"
      >
        <button
          className="btn"
          onClick={[MockFunction]}
          type="button"
        >
          <p
            className="button__title"
          >
            Confirm
          </p>
          <i />
        </button>
      </div>
    </div>
  </div>
</div>
`;
